lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -779,6 +813,260 @@ class ConvertAtenBmmOp : public OpConversionPattern<AtenBmmOp> {
 } // namespace
 
 namespace {
+struct QuantizationValues {
+  Value self;
+  Value zeroPoint;
+  bool isUnsigned;
+};
+
+QuantizationValues getQuantizationPerTensorValues(
+    ConversionPatternRewriter &rewriter, Location loc,
+    Aten_MakePerTensorQuantizedTensorOp makePerTensorQuantizedTensorOp,
+    const TypeConverter *const typeConverter) {
+  QuantizationValues values;
+  Value self = makePerTensorQuantizedTensorOp.getSelf();
+  Value zeroPoint = makePerTensorQuantizedTensorOp.getZeroPoint();
+  self = typeConverter->materializeTargetConversion(
+      rewriter, loc, typeConverter->convertType(self.getType()), self);
+  zeroPoint = typeConverter->materializeTargetConversion(
+      rewriter, loc, typeConverter->convertType(zeroPoint.getType()),
+      zeroPoint);
+  zeroPoint =
+      rewriter.create<arith::TruncIOp>(loc, rewriter.getI32Type(), zeroPoint);
+  auto torchDtype =
+      cast<ValueTensorType>(makePerTensorQuantizedTensorOp.getType())
+          .getDtype();
+
+  values.self = self;
+  values.zeroPoint = zeroPoint;
+  values.isUnsigned = torch_to_linalg::isUnsignedTorchType(torchDtype);
+
+  return values;
+}
+
+QuantizationValues getQuantizationPerChannelValues(
+    ConversionPatternRewriter &rewriter, Location loc,
+    Aten_MakePerChannelQuantizedTensorOp makePerChannelQuantizedTensorOp,
+    const TypeConverter *const typeConverter) {
+  QuantizationValues values;
+  Value self = makePerChannelQuantizedTensorOp.getSelf();
+  Value zeroPoint = makePerChannelQuantizedTensorOp.getZeroPoint();
+  self = typeConverter->materializeTargetConversion(
+      rewriter, loc, typeConverter->convertType(self.getType()), self);
+  zeroPoint = typeConverter->materializeTargetConversion(
+      rewriter, loc, typeConverter->convertType(zeroPoint.getType()),
+      zeroPoint);
+
+  // create a linalg op since we need to do some arithmetic on the zero point
+  // but is it a tensor.
+  RankedTensorType zeroPointType = cast<RankedTensorType>(zeroPoint.getType());
+  RankedTensorType selfType = cast<RankedTensorType>(self.getType());
+  if (zeroPointType.getElementTypeBitWidth() >
+      selfType.getElementTypeBitWidth()) {
+
+    zeroPoint = torch_to_linalg::createElementwiseLinalgGeneric(
+        rewriter, loc, zeroPoint, rewriter.getI32Type(),
+        [&](OpBuilder &b, Location loc, ValueRange payloadArgs) {
+          Value result = rewriter.create<arith::TruncIOp>(
+              loc, rewriter.getI32Type(), payloadArgs[0]);
+          b.create<linalg::YieldOp>(loc, result);
+        });
+  } else if (zeroPointType.getElementTypeBitWidth() <
+             selfType.getElementTypeBitWidth()) {
+    zeroPoint = torch_to_linalg::createElementwiseLinalgGeneric(
+        rewriter, loc, zeroPoint, rewriter.getI32Type(),
+        [&](OpBuilder &b, Location loc, ValueRange payloadArgs) {
+          Value result = rewriter.create<arith::ExtUIOp>(
```

Comment:
```suggestion
          Value result = rewriter.create<arith::ExtIOp>(
```
Since we assume it is an integer?